### PR TITLE
get_proposal_pos_embed of deformable_detr support return dtype of proposal 

### PR DESCRIPTION
**Pull request by DonggeunYu**
# What does this PR do?

Currently function `get_proposal_pos_embed` is fixed with return float32.
Modified to return dtype of proposal.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

- vision models: @amyeroberts

### CODE REVIEW

File: src/transformers/models/deformable_detr/modeling_deformable_detr.py [hunk 1]
```diff
@@ -1582,7 +1582,7 @@ def get_proposal_pos_embed(self, proposals):
         temperature = 10000
         scale = 2 * math.pi
 
-        dim_t = torch.arange(num_pos_feats, dtype=torch.int64, device=proposals.device).float()
+        dim_t = torch.arange(num_pos_feats, dtype=torch.int64, device=proposals.device).to(proposals.dtype)
```
Comment (amyeroberts) [inline src/transformers/models/deformable_detr/modeling_deformable_detr.py:1585]: Can't we pass in the dtype directly here? 

```suggestion
        dim_t = torch.arange(num_pos_feats, dtype=proposals.dtype, device=proposals.device)
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this!

File: src/transformers/models/deformable_detr/modeling_deformable_detr.py (hunk 1 above)
Comment (DonggeunYu) [reply src/transformers/models/deformable_detr/modeling_deformable_detr.py:1585]: Possible. Thank you

**Comment by DonggeunYu** [comment]
The lint problem is with other files.

**Comment by amyeroberts** [comment]
@DonggeunYu Indeed. I've tried re-running, but it doesn't appear to be using the new docker images (which it should). Could you try rebasing on main?

**Comment by DonggeunYu** [comment]
> @DonggeunYu Indeed. I've tried re-running, but it doesn't appear to be using the new docker images (which it should). Could you try rebasing on main?

@amyeroberts Done.

**Comment by DonggeunYu** [comment]
@amyeroberts Please check this PR.

**Review by amyeroberts — APPROVED** [review]
Thanks!

**Comment by amyeroberts** [comment]
@DonggeunYu Yesterday, we deprecated DETA. Could you rebase to include this change on main here? This will resolve the repo consistency check

**Comment by DonggeunYu** [comment]
> @DonggeunYu Yesterday, we deprecated DETA. Could you rebase to include this change on main here? This will resolve the repo consistency check

@amyeroberts Completed.

**Comment by amyeroberts** [comment]
@DonggeunYu My bad, the # Copied froms are still active. I've opened a PR to #31153 to remove these. Once merged, we can add that in and then we won't need to worry about DETA for this PR. Thanks for your patience!

**Comment by amyeroberts** [comment]
@DonggeunYu #31153 has been merged in. It would be good to still merge in this PR! I'll reopen. If you still wish to add and work on this, then adding upstream changes from main should fix the failing tests and I can merge. Otherwise fee lfree to close again :)

**Pull request closed without merging**